### PR TITLE
filePattern on FindSourceFiles is now nullable, and matches all files…

### PR DESCRIPTION
**Pull request by nmck257**
… on blank input, to make it a little more convenient to use in place of HasSourcePath

## Any additional context

Brief discussion over here https://github.com/openrewrite/rewrite/issues/3807#issuecomment-1850976790

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: rewrite-test/src/test/java/org/openrewrite/FindSourceFilesTest.java [hunk 1]
```diff
@@ -76,7 +77,8 @@ void windows() {
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {"hello.txt", "/hello.txt", "\\hello.txt", "./hello.txt", ".\\hello.txt"})
+    @ValueSource(strings = {"hello.txt", "/hello.txt", "\\hello.txt", "./hello.txt", ".\\hello.txt", ""})
+    @NullSource
     void findRoot(String filePattern) {
```
Comment (timtebeek) [inline rewrite-test/src/test/java/org/openrewrite/FindSourceFilesTest.java:82]: Two failures in this test it seems https://ge.openrewrite.org/s/akobtpkwdhh44/tests/overview?outcome=FAILED

Comment (nmck257) [reply rewrite-test/src/test/java/org/openrewrite/FindSourceFilesTest.java:82]: will check back over; was having issues with my local build the other day

Comment (nmck257) [reply rewrite-test/src/test/java/org/openrewrite/FindSourceFilesTest.java]: fixed; that's what I get for only skimming the test case I was adding on to :)

**Review by timtebeek — APPROVED** [review]

File: rewrite-test/src/test/java/org/openrewrite/FindSourceFilesTest.java (hunk 1 above)
Comment (timtebeek) [reply rewrite-test/src/test/java/org/openrewrite/FindSourceFilesTest.java]: All good, thanks for the quick fix!

**Comment by timtebeek** [comment]
Thanks again! Good to see this match-all option restored as compared to HasSourcePath before.

**Pull request merged**